src/main/java/ch/njol/skript/util/BlockDataUtils.java

```diff
@@ -0,0 +1,330 @@
+				
+				String stateString = blockDataString.replaceAll(".+\\[|]", "");
+				String[] splits = stateString.split(",");
+				for (String s : splits) {
+					String[] split = s.split("=");
+					String state = split[0];
+					String name = state.replace("_", " ");
+					
+					if (!stateTypes.containsKey(name))
+						stateTypes.put(name, new StateType(name, state));
+					
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Get all state types
+	 *
+	 * @return Collection of all state types
+	 */
+	public static Collection<StateType> getStateTypes() {
+		return stateTypes.values();
+	}
+	
+	/**
+	 * Get a string of all state type names
+	 * Used when registering class info
+	 *
+	 * @return String of all state type names
+	 */
+	public static String getStateTypeNames() {
+		return StringUtils.join(stateTypes.keySet(), ", ");
+	}
+	
+	/**
+	 * Get a state type by name
+	 *
+	 * @param name Name of state type
+	 * @return State type by name
+	 */
+	public static StateType getByName(String name) {
+		return stateTypes.get(name);
+	}
+	
+	/**
+	 * Change the state of a block
+	 *
+	 * @param block     Block to change
+	 * @param stateType StateType to change
+	 * @param value     New value
+	 */
+	public static void setBlockState(Block block, StateType stateType, Object value) {
+		BlockData blockData = block.getBlockData();
+		String key = block.getType().getKey().toString();
+		
+		try {
+			BlockData newData = Bukkit.createBlockData(key + "[" + stateType.type + "=" + value.toString().toLowerCase() + "]");
+			blockData = blockData.merge(newData);
+			block.setBlockData(blockData);
+		} catch (IllegalArgumentException ignore) {
+		}
+	}
+	
+	/**
+	 * Get a block state from a block
+	 *
+	 * @param block     Block to grab state from
+	 * @param stateType StateType to grab
+	 * @return Value of state type from block
+	 */
+	@Nullable
+	public static Object getBlockStateType(Block block, StateType stateType) {
+		BlockData blockData = block.getBlockData();
+		String blockDataString = blockData.getAsString();
+		
+		if (!blockDataString.contains("["))
+			return null;
+		
+		String stateString = blockDataString.replaceAll(".+\\[|]", "");
+		String[] splits = stateString.split(",");
+		for (String s : splits) {
+			String[] split = s.split("=");
+			String state = split[0];
+			String value = split[1];
+			if (!state.equalsIgnoreCase(stateType.name))
+				continue;
+			
+			if (NumberUtils.isNumber(value))
+				return NumberUtils.createNumber(value);
+			if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("false"))
+				return value.equalsIgnoreCase("true");
+			
+			@Nullable
+			StateValues stateValues = StateValues.getByName(value);
+			if (stateValues != null)
+				return stateValues;
+			
+			@Nullable
+			Direction direction = getDirection(value);
+			if (direction != null)
+				return direction;
+			return value;
+		}
+		return null;
+	}
+	
+	@Nullable
+	private static Direction getDirection(String dir) {
+		switch (dir) {
+			case "north":
```

Comment:
Should these string comparisons be done using hardcoded values in ``english.lang`` or something?

Reply:
Probably not, because these values aren't hard coded anywhere, they're just whatever the block data returns.